src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcVetRepositoryImpl.java

```diff
@@ -106,5 +107,31 @@ public Integer mapRow(ResultSet rs, int row) throws SQLException {
         return vets;
     }
 
+    @Override
+    public Vet findById(int id) {
+        List<Vet> vets = new ArrayList<>();
+        // Retrieve the list of all vets.
+        vets.addAll(this.jdbcTemplate.query(
+            "SELECT id, first_name, last_name FROM vets WHERE id =id",
+            BeanPropertyRowMapper.newInstance(Vet.class)));
+
+        // Retrieve the list of all possible specialties.
+        final List<Specialty> specialties = this.jdbcTemplate.query(
+            "SELECT id, name FROM specialties",
+            BeanPropertyRowMapper.newInstance(Specialty.class));
+
+        // Build each vet's list of specialties.
+        fillSpecialties(vets);
+        return vets.get(0);
+    }
+
+    @Override
+    public void save(Vet vet) {
+//        BeanPropertySqlParameterSource parameterSource = new BeanPropertySqlParameterSource(vet);
```

Comment:
Look at https://github.com/cgerard321/STQA-pet-clinic-F2020/blob/master/src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcPetRepositoryImpl.java#L89 for an implementation

Also add unit tests.